src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj

```diff
@@ -31,13 +31,8 @@
   <ItemGroup>
     <ProjectReference
       Include="$(RepoRoot)src/coreclr/tools/aot/crossgen2/crossgen2_publish.csproj"
-      AdditionalProperties="_IsPublishing=true
-        ;RuntimeIdentifier=$(PackageRID)
-        ;NativeAotSupported=$(NativeAotSupported)
-        ;CoreCLRArtifactsPath=$(CoreCLRArtifactsPath)
-        ;ObjCopyName=$(ObjCopyName)
-        ;R2ROverridePath=$(MSBuildThisFileDirectory)ReadyToRun.targets"
-      Targets="Publish;PublishItemsOutputGroup"
+      AdditionalProperties="CoreCLRArtifactsPath=$(CoreCLRArtifactsPath)
+        ;ObjCopyName=$(ObjCopyName)"
```

Comment:
The problem with additional properties is that they flow to the entire referenced project graph. I.e. with https://github.com/dotnet/runtime/pull/91468, crossgen2_publish.csproj now references ILLink.Tasks.csproj which references Mono.Linker.csproj with those global properties set. Other build parts reference these projects as well, but without these global properties set, which results in the following error:

> /Users/runner/work/1/s/.dotnet/sdk/8.0.100-rc.1.23415.11/Sdks/Microsoft.NET.Sdk/targets/Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1047: Assets file '/Users/runner/work/1/s/artifacts/obj/Mono.Linker/project.assets.json' doesn't have a target for 'net8.0/osx-arm64'. Ensure that restore has run and that you have included 'net8.0' in the TargetFrameworks for your project. You may also need to include 'osx-arm64' in your project's RuntimeIdentifiers. [/Users/runner/work/1/s/src/tools/illink/src/linker/Mono.Linker.csproj]

Reply:
With this change, we don’t pass the RID through AdditionalProperties and these properties shouldn’t influence anything in the restore process, so hopefully that fixes this problem.

Reply:
These additional global properties still flow to the linker projects which cause double builds, double writes and in the worst case races. We should avoid that.

Reply:
I'll see if I can get rid of the rest of them.